src/utils/inngest/functions.ts

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios';
+import { inngest } from './client';
+import { Database } from '../database.types';
+import { createClient } from '@supabase/supabase-js';
+import { getTxCountBatch } from '../claims/txHistoryCheck';
+import { Networks } from '../database.enums';
+import { providers } from '../ethereum';
+import { ethers } from 'ethers';
+import { get } from 'http';
+
+const supabase = createClient<Database>(
+  process.env.SUPABASE_URL as string,
+  process.env.SUPABASE_ANON_KEY as string
+);
+
+const BatchSize = parseInt(process.env.BATCH_SIZE as string) || 50;
+const queryBatchSize = 1000;
+export const userTxCount = inngest.createFunction(
+  { id: 'user-tx-count' },
+  { event: 'events/user-tx-count' },
+  async ({ event, step }) => {
+    const users: { user_address: string; guild_id: string }[] = [];
+    let uRange: string[] = [];
+    let iter: number = 0;
+    while (true && iter < 20) {
+      const uRange = await step.run('fetch guild users', async () => {
+        const { data, error } = await supabase
+          .from('guild_member_configuration')
+          .select('user_address,guild_id')
+          .eq('game_id', event.data.gameId as number)
+          .range(iter * queryBatchSize, (iter + 1) * queryBatchSize - 1);
+        if (error) {
+          return [];
+        }
+        return data;
+      });
+      users.push(...uRange);
+      if (uRange.length < queryBatchSize) {
+        break;
+      }
+      iter++;
+    }
+    // console.log('users:', users.length);
+    //await step.sleep('wait-a-moment', '1 second');
+    let chunks = chunkArray(users, BatchSize);
+
+    let txCounts: any[] = [];
+    for (var i = 0; i < chunks.length; i++) {
+      const t = await step.run('get-tx-count', async () => {
+        const p = providers[Networks.networks_base_mainnet];
+        const reqs = chunks[i].map((u, idx) => {
+          return {
+            id: idx,
+            jsonrpc: '2.0',
+            method: 'eth_getTransactionCount',
+            params: [u.user_address, 'latest'],
+          };
+        }) as Array<{
+          id: number;
+          jsonrpc: '2.0';
+          method: string;
+          params: Array<any>;
```

Comment:
define this as type outside of map function

Reply:
this is unused outside of the scope